src/Compilers/CSharp/Test/Semantic/Semantics/IndexAndRangeTests.cs

```diff
@@ -695,7 +695,7 @@ public class C {
             comp.VerifyDiagnostics(
                 // (11,17): error CS0656: Missing compiler required member 'System.Runtime.CompilerServices.RuntimeHelpers.GetSubArray'
                 //         var x = arr[0..2];
-                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "arr[0..2]").WithArguments("System.Runtime.CompilerServices.RuntimeHelpers", "GetSubArray").WithLocation(11, 17));
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "0..2").WithArguments("System.Runtime.CompilerServices.RuntimeHelpers", "GetSubArray").WithLocation(11, 21));
```

Comment:
>Diagnostic [](http://example.com/codeflow?start=16&length=10)

It looks like we are lacking test coverage for the following scenarios affected by the change:
- pointer indexing
- array creation
- variable declaration with dimensions on the type

Reply:
- Will add a test for pointer indexing.

- Array creation is covered in `CS0266ERR_NoImplicitConvCast13:

https://github.com/dotnet/roslyn/blob/67c6bed1581a0d5db35b7e68d6766560a089ea1a/src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs#L9235-L9244

- If you mean `int[o] arr;`, then this will produce `error CS0270: Array size cannot be specified in a variable declaration (try initializing with a 'new' expression)`.

Reply:
> If you mean int[o] arr;, then this will produce error CS0270: Array size cannot be specified in a variable declaration (try initializing with a 'new' expression).

Do we have an existing test demonstrating this with type mismatch in the illegal sizes?

Reply:
@AlekseyTs If you mean something like:

```
    public void M(object o) {
        int[o] x;
    }
```

Then I think there is no test for it. I'll add it once you confirm this is the case you're requesting.

Reply:
> If you mean something like ...

Yes, something like this. We want a test that hits your changes in BindArrayDimension for a variable declaration